download-basket/src/main/java/downloadbasket/helpers/SendDownloadDetailsToEmailThread.java

```diff
@@ -0,0 +1,289 @@
+	private final String PARAM_CROPPING_MODE = "croppingMode";
+	private final String PARAM_CROPPING_LAYER = "croppingLayer";
+	private final String PARAM_LAYER = "layer";
+	private final String PARAM_LAYER_ID = "id";
+
+	/**
+	 * Constructor.
+	 * 
+	 * @param downLoadDetails
+	 *            download details
+	 * @param userDetails
+	 *            user details
+	 */
+	public SendDownloadDetailsToEmailThread(JSONArray downLoadDetails, JSONObject userDetails, Locale locale) {
+		this.downLoadDetails = downLoadDetails;
+		this.userDetails = userDetails;
+		this.locale = locale;
+	}
+
+	/**
+	 * 
+	 * Overrides the run method. Collects the download materials and sends them
+	 * using the variables given in constructor.
+	 * 
+	 */
+	@Override
+	public void run() {
+
+		try {
+			DownloadServices ds = new DownloadServices();
+			ArrayList<ZipDownloadDetails> mergeThese = new ArrayList<ZipDownloadDetails>();
+			final String strTempDir = PropertyUtil.get("oskari.wfs.download.folder.name");
+			String normalWayDownload = PropertyUtil.get("oskari.wfs.download.normal.way.downloads");
+			String[] temp = normalWayDownload.split(",");
+			NormalWayDownloads normalDownloads = new NormalWayDownloads();
+			for (int i = 0; i < temp.length; i++) {
+				normalDownloads.addDownload(temp[i]);
+			}
+
+			for (int i = 0; i < downLoadDetails.length(); i++) {
+				JSONObject download = downLoadDetails.getJSONObject(i);
+				final String croppingMode = download.getString(PARAM_CROPPING_MODE);
+				String croppingLayer = "";
+				if (download.has(PARAM_CROPPING_LAYER)) {
+					croppingLayer = download.getString(PARAM_CROPPING_LAYER);
+				}
+
+				LoadZipDetails ldz = new LoadZipDetails();
+				ldz.setTemporaryDirectory(strTempDir);
+				ldz.setUserEmail(userDetails.getString("email"));
+				ldz.setLanguage(this.locale.getLanguage());
+				ldz.setDownloadNormalWay(normalDownloads.isBboxCropping(croppingMode, croppingLayer));
+				OskariLayerService mapLayerService = new OskariLayerServiceIbatisImpl();
+				OskariLayer oskariLayer = mapLayerService.find(download.getString(PARAM_LAYER_ID));
+				String srs = "EPSG:4326";
+				if (oskariLayer != null) {
+					srs = oskariLayer.getSrs_name();
+				}
+
+				if (ldz.isDownloadNormalWay()) {
+					ldz.setGetFeatureInfoRequest(OGCServices.getFilter(download, true));
+					ldz.setWFSUrl(OGCServices.doGetFeatureUrl(srs, download, false));
+				} else {
+
+					ldz.setGetFeatureInfoRequest("&filter=" +OGCServices.getPluginFilter(download));
+					ldz.setWFSUrl(OGCServices.doGetFeatureUrl(srs, download, true));
+				}
+
+				final String fileLocation = ds.loadZip(ldz, this.locale);
+
+				if (fileLocation != null) {
+					ZipDownloadDetails zdd = new ZipDownloadDetails();
+					zdd.setFileName(fileLocation);
+					final String sLayer = Helpers.getLayerNameWithoutNameSpace(download.getString(PARAM_LAYER));
+					zdd.setLayerName(sLayer);
+					mergeThese.add(zdd);
+				}
+			}
+
```

Comment:
Can you split this method to multiple parts? The indentation goes to 8 tabs/32 spaces and it's kinda long method doing a bunch of different things.